### PR TITLE
New Sample - app-feed-members - Nodejs

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**

### CODE REVIEW

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                var member = await TeamsInfo.getMember(context, memberId);
+                await context.sendActivity(`Welcome to the team :${member.name}`);
+                await this.listMembersAsync(context, members);
+                
+            await next();
+        });
+    }
+
+    async onInstallationUpdateActivity(context) {
+        if (context.activity.conversation.conversationType === 'channel') {
```
Comment (rm3f) [inline samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:44]: I think the requirement specified by Tatiana was to return the information about the user who installed the bot/app if possible. It is likely the activity.from will have the id of the user. Then you can use that to call get member and provide all info about that user including tenant id. 

Similar logic should be added on member added event, when you will display the info returned by get member for the member that was added.

the logic with the list of all members can happen when the user send the list command

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            await context.sendActivity(MessageFactory.text(`Welcome to Microsoft Teams type list by @ mention the bot to get list of all members.`));
+        } else {
+            await context.sendActivity(MessageFactory.text('Welcome to Microsoft Teams type list by @ mention the bot to get list of all members.'));
```
Comment (rm3f) [inline samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:47]: this IF statement has the same code on both conditions

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                await this.listMembersAsync(context, members);
+            } else {
+                await this.cardActivityAsync(context, false);
+            }
+
+            await next();
+        });
+
+        this.onMembersAdded(async (context, next) => {
+                const members = await TeamsInfo.getPagedMembers(context);
```
Comment (rm3f) [inline samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:33]: should not get all members here, use the member id returned by the event

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js (hunk 1 above)
Comment (Prithvi-MSFT) [reply samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:44]: Currently the bot is supported for group chat scope which returns list of members whenever app is installed in group chat/ new member added or list command is given. Do we have to change this scenario ?

Comment (rm3f) [reply samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:44]: On installation update, lines, 43-49 i see a welcome message being sent. Am i looking at the wrong things? 

async onInstallationUpdateActivity(context) {
        if (context.activity.conversation.conversationType === 'channel') {
   await context.sendActivity(MessageFactory.text(`Welcome to Microsoft Teams type list by @ mention the bot to get list of all members.`));
        } else {
            await context.sendActivity(MessageFactory.text('Welcome to Microsoft Teams type list by @ mention the bot to get list of all members.'));      }
    }

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js (hunk 2 above)
Comment (rm3f) [reply samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:47]: My understanding of requirements is: 
On member added -  display full info of the member added by calling get-member based on the id returned by the member added event. 
On installation update we should display the detailed  info of the user that installed the app (calling get member) and possibly also the info for all the other users   in the  chat by calling get-members-paged
Optionally, when user sends "list" command, bot should return the list of all users.

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js (hunk 1 above)
Comment (rm3f) [reply samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:44]: My understanding of requirements is:
On member added - display full info of the member added by calling get-member based on the id returned by the member added event.
On installation update we should display the detailed info of the user that installed the app (calling get member) and possibly also the info for all the other users in the chat by calling get-members-paged
Optionally, when user sends "list" command, bot should return the list of all users.

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js (hunk 3 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:33]: Updated

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js (hunk 2 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:47]: Updated

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js (hunk 1 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-feed-members/nodejs/bots/teamsConversationBot.js]: Updated

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                        type: "TextBlock",
+                        text: "Here are your details:",
+                        weight: "Bolder",
+                        size: "Large",  // Increased size
+                        spacing: "Large"  // Added spacing
+                    },
+                    {
+                        type: "FactSet",
+                        facts: [
+                            { title: "Name:", value: member.name },
```
Comment (rm3f) [inline samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:67]: The tenant Id needs to be included in the information about the user, as it is the main difference about federated scenarios. That is the way to know from which tenant each of the members of the group chat are from, and to prove the federated scenario.

Comment (Harikrishnan-MSFT) [reply samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:67]: Updated.

image

**Comment by Pawank-MSFT** [comment]
Hi @rm3f We have fixed the comments please check and confirm.

**Review by rm3f — APPROVED** [review]

File: samples/bot-feed-members/nodejs/bots/teamsConversationBot.js (hunk 4 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-feed-members/nodejs/bots/teamsConversationBot.js:67]: Updated

**Comment by Harikrishnan-MSFT** [comment]
Build Pipeline - 
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/10614532106

image

File: README.md [hunk 5]
```diff
@@ -114,6 +114,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |23| Bot configuration app with type ahead search control.|This sample demonstrates the features of bot configuration and reconfiguration for both teams and group chats, including a type-ahead search (static and dynamic) control on Adaptive Cards.|[View][BotConfigurationApp#csharp]|[View][BotConfigurationApp#nodejs
 |23| Bot configuration app with auth |This sample demonstrates the features of bot configuration auth and reconfiguration for both teams and group chats.|[View][BotConfigurationAppAuth#csharp]|[View][BotConfigurationAppAuth#nodejs]
 |24| Tag mention bot  | This sample app demonstrates the use of tag mention funtionality in teams scope using Bot Framework. |[View][Tagmention#csharp]|[View][Tagmention#nodejs] ![toolkit-icon](assets/toolkit-icon.png) | | |
+|25| Apps in Federated | This sample showcases how apps function within federated group chats. ||[View][bot-feed-members#js] ![toolkit-icon](assets/toolkit-icon.png)
```
Comment (Pawank-MSFT) [inline README.md:117]: Need to change description based on below documnet
https://review.learn.microsoft.com/en-us/microsoftteams/platform/concepts/build-and-test/shared-channels?branch=pr-en-us-10792&tabs=bot

File: samples/bot-feed-members/nodejs/README.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+- nodejs
+extensions:
+ contentType: samples
+ createdDate: "08-21-2024 20:15:25"
+urlFragment: officedev-microsoft-teams-samples-bot-feed-members-nodejs
+---
+
+# Apps in federated
+
+Apps in federated group chats.
```
Comment (Pawank-MSFT) [inline samples/bot-feed-members/nodejs/README.md:18]: Add proper description
https://review.learn.microsoft.com/en-us/microsoftteams/platform/concepts/build-and-test/shared-channels?branch=pr-en-us-10792&tabs=bot

**Review by Pawank-MSFT — COMMENTED** [review]
Just small comment please fix

File: samples/bot-feed-members/nodejs/README.md (hunk 6 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-feed-members/nodejs/README.md:18]: Fixed

File: README.md (hunk 5 above)
Comment (Harikrishnan-MSFT) [reply README.md:117]: Fixed

**Review by Pawank-MSFT — APPROVED** [review]
looks correct, approving!

**Review by Mohammed-MSFT — APPROVED** [review]

**Review by Harikrishnan-MSFT — APPROVED** [review]
Looks Good, approving!

**Pull request merged**